content/articles/how-to-debug-a-nodejs-application-running-in-a-docker-container/index.md

```diff
@@ -0,0 +1,436 @@
+### Introduction
+Debugging an application involves detecting and removing existing and potential errors also known as bugs in a code which can cause it to work unexpectedly or even crash.
+
+Debugging allows the developers to find and resolve bugs or defects in the application to prevent incorrect and unwanted operations.
+
+In this tutorial, we will build a Node.js application with a bug in it, run it in a Docker container and use Visual Studio Code debugging tools for Node.js to debug it. The tutorial can be used by developers as a reference when building their applications.
+
+Developers and the readers ought to have prior knowledge of [JavaScript programming](https://www.javascript.com/learn/) and Docker containers.
+
+### Prerequisites
+- Prior knowledge of [Docker containers](https://www.docker.com/resources/what-container) and [Node.js](https://nodejs.dev/learn).
+- [Docker](https://www.docker.com/products/docker-desktop) installed.
+- [Node.js](https://nodejs.org/en/) installed.
+- [Microsoft Visual Studio Code](https://code.visualstudio.com/download) installed.
+- A working web browser, preferably [Google Chrome](https://www.google.com/chrome).
+
+### Initializing the to-do application
+We will create a simple to-do list application that allows the users to add and delete tasks. We will also introduce a minor bug when coding the application and use Visual Studio Code to address the issue and fix it.
+
+The knowledge gained in this tutorial will help the reader be able to debug their applications.
+
+#### Step 1
```

Comment:
Please add headings to these steps. eg `Step 1 -- Create directory`. If this isn't possible, consider removing them, and join these parts with words like `Then, ...`, `Now, ...`